hamilflow/models/particle_in_box.py

```diff
@@ -0,0 +1,143 @@
+from functools import cached_property
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+from pydantic import BaseModel, Field, computed_field, field_validator
+
+
+PLANCK_CONST = 6.626e-34
+PI = 3.142
+
+
+class FiniteBox(BaseModel):
+    r"""Definition of the potential of the one dimensional box
+
+    For consistency, we always use $\mathbf x$ for displacement,
+    $L$ for box size.
+
+    The potential distribution we use is
+
+    $$
+    V(x)=\left\{\begin{array}{ll}
+    -V_{0} & |x|<L \\
+    0 & |x|>L
+    \end{array}\right.
+    $$
+
+    :cvar length: length of the box
+    :cvar p_height: potential well height
+    """
+
+    length: float = Field(ge=0)
```

Comment:
`ge=0.0` for being a `float`.

Also the `length` you defined here is actually _half-width_. It may become tricky at some point..